src/main/java/SpaceFilter.java

```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
+
+/**
+ * This class represents a filter. It works by checking a given schedule for the given criteria.
```

Comment:
This is more a class comment for `Filter` than for this class specifically, so I think u should move it there